### ISSUE TITLE
Clean up notDisposed leaks in Flutter Framework, phase 1.

### ISSUE THREAD

**Issue by polina-c** [dart-lang/leak_tracker#134]
Looking for volunteers to help with this!

Subtask of: https://github.com/flutter/devtools/issues/3951

Issues: [all](https://github.com/pulls?q=is%3Aissue+label%3A%22a%3A+leak+tracking%22+), [open](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3A%22a%3A+leak+tracking%22).
PRs: [all](https://github.com/pulls?q=is%3Apr+label%3A%22a%3A+leak+tracking%22+), [open](https://github.com/pulls?q=is%3Aopen+is%3Apr+archived%3Afalse+label%3A%22a%3A+leak+tracking%22+).

All invocations of `testWidgets` under [flutter/test](https://github.com/flutter/flutter/tree/master/packages/flutter/test)  (~ 10,000) should be converted to [testWidgetsWithLeakTracking](https://pub.dev/packages/leak_tracker_flutter_testing#how-to-use-the-helpers) by iterating the steps:

1. Comment on this issue what subset of tests you plan to work on so that we do not duplicate work. Then delete the added comment as soon as you are done or if your plans changed. If all tests seems to be occupied, let me know, and I will redistribute them.

2. Convert set of `testWidgets`, create PR and, if tests pass, request review and merge. Example: https://github.com/flutter/flutter/pull/132806

3. For tests that fail, investigate them and search for the culprit class name in onen issues or PRs (check links above). If the issue for this concrete leak is already created or PR is on the way, leave this test to be 'testWidgets'. Otherwise, move the conversion to a separate PR, and do one of two in this PR:

    a. Fix the failure if it is strait-forward.

    b. Create issue that explains the leak, create allowlist, and link the issue near the allowlist. Example: https://github.com/flutter/flutter/pull/130468. Then either allow triage process to happen or assign the issue to yourself and resolve it in a separate PR.

4. Link the issue and the fix in a comment for this PR.

5. Do not:
    - Instrument new disposables, because we want to clean up leaks first for already instrumented ones
    - Convert many tests with the same issue linked to allowlists. One test is enough to repro the issue.

Focus areas: 
@ksokolovskyi: test/widgets
@droidbg: test/paintings, test/rendering and test/gestures
@NobodyForNothing: paused (thank you for what is done and in process)
@polina-c: coordinating complicated cases and [integration with testWidgets](https://docs.google.com/document/d/1bXannXQ8rEtfwxEVd7LDo7xiVu5-QxRoall28MIIFss/edit#heading=h.dk8qogiupft4)

Discord calls: [1](https://discord.com/channels/608014603317936148/608014603317936150/1146232649669693460), [2](https://discord.com/channels/608014603317936148/608014603317936150/1150172011830706331)
[Thread on discord for leak_tracker updates](https://discord.com/channels/608014603317936148/1106667330093723668/1154901557171732510 )

**Comment by polina-c** [comment]
- [ ] cupertino
        - [ ] material (~ 4000 tests - [failures](https://raw.githubusercontent.com/flutter/flutter/6a57468cd4e29e900d3fca1f931e023af9c546df/packages/flutter/test/material/_material_test_results.txt))
            - [x] https://github.com/flutter/flutter/issues/130467 
            - [x] https://github.com/flutter/flutter/issues/130354
            - [ ] https://github.com/flutter/flutter/issues/130469
            - [x] https://github.com/flutter/flutter/issues/132620
            - [x] https://github.com/flutter/flutter/issues/132769
            - [ ] https://github.com/flutter/flutter/issues/133071
            - [ ] https://github.com/flutter/flutter/pull/133352
        - [ ] painting
        - [ ] physics
        - [ ] rendering
        - [ ] scheduler
        - [ ] semantics
        - [ ] services
        - [ ] widgets

**Comment by polina-c** [comment]
progress check: 2071 tests are converted

**Comment by polina-c** [comment]
Yes, not al extenders of ChangeNotifier do this at the moment.
For now it is added as needed, so no tracking yet. I have two PRs for this waiting lgtm:
https://github.com/flutter/flutter/pull/133716
https://github.com/flutter/flutter/pull/133717

Feel free just add it where you need.

**Comment by polina-c** [comment]
> @polina-c Where do we track that all required widgets/services do
> 
> ```
>     if (kFlutterMemoryAllocationsEnabled) {
>       maybeDispatchObjectCreation();
>     }
> ```
> 
> in their constructor?

Yes, not al extenders of ChangeNotifier do this at the moment.
For now it is added as needed, so no tracking yet. I have two PRs for this waiting lgtm:
https://github.com/flutter/flutter/pull/133716
https://github.com/flutter/flutter/pull/133717

Feel free to just add it where you need.

**Comment by ksokolovskyi** [comment]
@polina-c Could you please take a look at this PR, which adds tracking to the ScrollController: https://github.com/flutter/flutter/pull/133759 ?

**Comment by polina-c** [comment]
progress check: 3458 tests covered

**Comment by polina-c** [comment]
@ksokolovskyi , I updated description asking to communicate subset of tests where you plan to work.

I am working on test/material and test/painting.

**Comment by ksokolovskyi** [comment]
@polina-c, nice, thanks.
If you don't mind, I would like to continue with test/widgets.

**Comment by NobodyForNothing** [comment]
I am starting work on test/services if that's fine with you.

**Comment by polina-c** [comment]
progress check: 5223 tests are converted

**Comment by NobodyForNothing** [comment]
working on test/semantics

**Comment by polina-c** [comment]
Progress check: 6202 tests are converted.

**Comment by polina-c** [comment]
Labelled issues and PRs with "a: leak tracking" and linked in description.

**Comment by polina-c** [comment]
@ksokolovskyi , @NobodyForNothing , I am participating internal event on the week of Sep 25. I will handle all PRs, when I am back on Oct 2.

**Comment by droidbg** [comment]
@polina-c I also like to contribute to this, which subset are still available?

**Comment by polina-c** [comment]
> @polina-c I also like to contribute to this. Can I start with test/paintings and test/gestures, if it is not taken by anyone?

@droidbg, yes and thank you!
Added test/paintings and test/gestures to the focus areas in description.
And, please, take into account this issue and related PRs working with gestures: https://github.com/flutter/flutter/issues/136132

**Comment by polina-c** [comment]
covered: 8826
remaining: 1036.

**Comment by droidbg** [comment]
@polina-c I have raised PRs for test/gesture and test/painting, could you please review it. 
 
Also, test/gesture is done and test/painting is almost done, so I will be picking test/rendering in parallel now.

**Comment by droidbg** [comment]
@polina-c I have raised PRs for test/rendering, could you please review it. 
Also, can I pick test/cupertino if no one is working on it now?

**Comment by polina-c** [comment]
Will appreciate votes here: https://github.com/flutter/devtools/issues/6266#issuecomment-1758474605

@NobodyForNothing @ksokolovskyi @droidbg @navaronbracke

**Comment by polina-c** [comment]
FYI, update on leak_tracker progress: https://discord.com/channels/608014603317936148/1106667330093723668/1162097704462196788

**Comment by droidbg** [comment]
Hello folks, these are the updates on the subsets I was working on: 
- test/gesture - done. 
- test/painting - 1 PR is [open](https://github.com/flutter/flutter/pull/136170) , rest are done.
- test/rendering - done. 
- test/cupertino - 1 PR is [open](https://github.com/flutter/flutter/pull/135355) by NobodyForNothing , rest are done. 

@polina-c 
Can I pick up test/material, if no one is working on it? 
or if there is anything else that I can help with, do let me know. Thanks.

**Comment by ksokolovskyi** [comment]
Hi @droidbg, thanks for your work!
Currently, I'm responsible for material and widgets tests.
All material tests will be covered after [#137004](https://github.com/flutter/flutter/pull/137004) merge.
@polina-c, do you have any tasks for @droidbg?

**Comment by polina-c** [comment]
@ksokolovskyi, @droidbg , @NobodyForNothing , I greatly appreciate your help!

Closing this issue in favor of the next phase, where help is also needed and appreciated: https://github.com/flutter/flutter/issues/137311

**Comment by polina-c** [comment]
> do you have any tasks for @droidbg?

Yes, pick any class or test-set in https://github.com/flutter/flutter/issues/137311
